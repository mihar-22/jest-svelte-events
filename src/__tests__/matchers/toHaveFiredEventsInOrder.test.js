import '../../'
import { render, fireEvent } from '@testing-library/svelte'
import Comp from '../fixtures/Comp.svelte'

describe('.toHaveFiredEventsInOrder', () => {
  test('passes when given events that fired in correct order', async () => {
    const { component, getByText } = render(Comp)
    listen(component, ['e1', 'e2'])
    const e1Button = getByText('e1')
    const e2Button = getByText('e2')
    await fireEvent.click(e1Button)
    await fireEvent.click(e2Button)
    await fireEvent.click(e1Button)
    expect(component).toHaveFiredEventsInOrder(['e1', 'e2', 'e1'])
  })

  test('fails when given events that did not fire in correct order', async () => {
    const { component, getByText } = render(Comp)
    listen(component, ['e1', 'e2'])
    const e1Button = getByText('e1')
    const e2Button = getByText('e2')
    await fireEvent.click(e2Button)
    await fireEvent.click(e1Button)
    expect(() => {
      expect(component).toHaveFiredEventsInOrder(['e1', 'e2'])
    }).toThrowErrorMatchingSnapshot()
  })
})

describe('.not.toHaveFiredEventsInOrder', () => {
  test('passes when given events that did not fire in correct order', async () => {
    const { component, getByText } = render(Comp)
    listen(component, ['e1', 'e2'])
    const e1Button = getByText('e1')
    const e2Button = getByText('e2')
    await fireEvent.click(e2Button)
    await fireEvent.click(e1Button)
    expect(component).not.toHaveFiredEventsInOrder(['e1', 'e2'])
  })

  test('fails when given events that did fire in correct order', async () => {
    const { component, getByText } = render(Comp)
    listen(component, ['e1', 'e2'])
    const e1Button = getByText('e1')
    const e2Button = getByText('e2')
    await fireEvent.click(e1Button)
    await fireEvent.click(e2Button)
    expect(() => {
      expect(component).not.toHaveFiredEventsInOrder(['e1', 'e2'])
    }).toThrowErrorMatchingSnapshot()
  })
})
