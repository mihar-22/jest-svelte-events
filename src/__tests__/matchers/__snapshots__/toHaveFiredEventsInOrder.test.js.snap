// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.not.toHaveFiredEventsInOrder fails when given events that did fire in correct order 1`] = `
"[2mexpect([22m[31mcomponent[39m[2m).[22mnot[2m.[22mtoHaveFiredEventsInOrder[2m([22m[32mexpected[39m[2m)[22m

Expected not to have fired in order: [32m[\\"e1\\", \\"e2\\"][39m
Recevied events in order: [31m[\\"e1\\", \\"e2\\"][39m"
`;

exports[`.toHaveFiredEventsInOrder fails when given events that did not fire in correct order 1`] = `
"[2mexpect([22m[31mcomponent[39m[2m).[22mtoHaveFiredEventsInOrder[2m([22m[32mexpected[39m[2m)[22m

Expected to have fired in order: [32m[\\"e1\\", \\"e2\\"][39m
Recevied events in order: [31m[\\"e2\\", \\"e1\\"][39m"
`;
